From 19afd2a36446ce45aa8037541ba98f3f612bb3ea Mon Sep 17 00:00:00 2001
From: chdugw_auto <jkulkarni@maxlinear.com>
Date: Tue, 31 Aug 2021 09:38:21 +0300
Subject: [PATCH] Merge pull request #3505 in SW_UGW/linux from
 feature/DRVLIB_SW-4232-prx-driver-add-field-dp-en to master

* commit '3e431a30d5d97a5beba1f6e701f93155a486a342':
  DRVLIB_SW-4232: Update CQM DP intf cbm_dp_en_data struct
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 1 +
 include/net/datapath_api.h                   | 1 +
 include/net/lantiq_cbm_api.h                 | 6 +++++-
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index 776fb8bae8c2..74fd7460412e 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -2482,6 +2482,7 @@ static void fill_tx_ring_data(struct cbm_dp_alloc_complete_data *dp_data)
 		dp_data->tx_ring[ring_idx]->policy_num = 1;
 		dp_data->tx_ring[ring_idx]->tx_pkt_size = 0;
 		dp_data->tx_ring[ring_idx]->tx_poolid = 0;
+		dp_data->tx_ring[ring_idx]->tx_deq_port = dp_data->deq_port;
 	}
 }
 
diff --git a/include/net/datapath_api.h b/include/net/datapath_api.h
index 5bdf1a1ff02d..948d9c9458a2 100644
--- a/include/net/datapath_api.h
+++ b/include/net/datapath_api.h
@@ -1339,6 +1339,7 @@ struct dp_tx_ring {
 				  * Note, this setting is not valid for
 				  *GRX500/FLM
 				  */
+	u32 tx_deq_port; /*!< [out] cqm deque port id for this txin ring */
 };
 
 /*! @brief struct dp_dev_data, which used for DirectConnected (DC)
diff --git a/include/net/lantiq_cbm_api.h b/include/net/lantiq_cbm_api.h
index 095c103afa34..fd0ee5e56aa0 100644
--- a/include/net/lantiq_cbm_api.h
+++ b/include/net/lantiq_cbm_api.h
@@ -384,7 +384,11 @@ struct cbm_dp_alloc_data {
 struct cbm_dp_en_data {
 	int dp_inst; /*input*/
 	int cbm_inst; /*input*/
-	u32 deq_port; /*input: -1 means not valid*/
+	union {
+		u32 deq_port; /*input: -1 means not valid*/
+		u32 deq_port_list[8]; /*input: -1 means not valid*/
+	};
+	int num_deq_port; /* input: number of dequeue port */
 	u32 dma_chnl_init;/*input :0 - no init, 1:- init DMA channel*/
 	/**
 	 *[in] flag to indicate whether need new
